logical/framework/openapi.go

```diff
@@ -148,15 +148,19 @@ type OASMediaTypeObject struct {
 }
 
 type OASSchema struct {
-	Type        string                `json:"type,omitempty"`
-	Description string                `json:"description,omitempty"`
-	Properties  map[string]*OASSchema `json:"properties,omitempty"`
-	Items       *OASSchema            `json:"items,omitempty"`
-	Format      string                `json:"format,omitempty"`
-	Pattern     string                `json:"pattern,omitempty"`
-	Enum        []interface{}         `json:"enum,omitempty"`
-	Example     interface{}           `json:"example,omitempty"`
-	Deprecated  bool                  `json:"deprecated,omitempty"`
+	Type             string                `json:"type,omitempty"`
+	Description      string                `json:"description,omitempty"`
+	Properties       map[string]*OASSchema `json:"properties,omitempty"`
+	Items            *OASSchema            `json:"items,omitempty"`
+	Format           string                `json:"format,omitempty"`
+	Pattern          string                `json:"pattern,omitempty"`
+	Enum             []interface{}         `json:"enum,omitempty"`
+	Example          interface{}           `json:"example,omitempty"`
+	Deprecated       bool                  `json:"deprecated,omitempty"`
+	Required         bool                  `json:"required,omitempty"`
+	DisplayName      string                `json:"x-vault-display-name,omitempty" mapstructure:"x-vault-display-name,omitempty"`
```

Comment:
For my own education, I'm just curious why these fields begin with `x-vault-`. Is it because they map to header fields in some cases?

Reply:
The non `x-vault-*` fields are official OpenAPI spec fields with defined meanings. But the spec allows for arbitrary `x-...` extensions. These new fields don't really have good equivalents in OpenAPI, hence the 'x-vault'. For Sensitive strings, there _is_ a defined format of "password", but I don't think our sensitive fields necessarily will be strings, nor passwords.